src/NovelRT.Interop/Windowing/NrtIWindowingDevice.cpp

```diff
@@ -0,0 +1,80 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root
+// for more information.
+
+#include "../LifetimeExtender.h"
+#include <NovelRT.Interop/NrtErrorHandling.h>
+#include <NovelRT.Interop/Windowing/NrtIWindowingDevice.h>
+#include <NovelRT/Windowing/Windowing.h>
+
+using namespace NovelRT::Interop;
+using namespace NovelRT::Windowing;
+
+NrtResult Nrt_IWindowingDevice_Destroy(NrtIWindowingDeviceHandle device)
+{
+    if (device == nullptr)
+    {
+        Nrt_setErrIsNullInstanceProvidedInternal();
+        return NRT_FAILURE_NULL_INSTANCE_PROVIDED;
+    }
+
+    if (Lifetime::Release(reinterpret_cast<IWindowingDevice*>(device)))
+    {
+        return NRT_SUCCESS;
+    }
+    else
+    {
+        Nrt_setErrMsgIsAlreadyDeletedOrRemovedInternal();
+        return NRT_FAILURE_ALREADY_DELETED_OR_REMOVED;
+    }
+}
+
+NrtBool Nrt_IWindowingDevice_GetIsVisible(NrtIWindowingDeviceHandle device)
+{
+    return reinterpret_cast<IWindowingDevice*>(device)->GetIsVisible();
+}
+
+NrtBool Nrt_IWindowingDevice_GetShouldClose(NrtIWindowingDeviceHandle device)
+{
+    return reinterpret_cast<IWindowingDevice*>(device)->GetShouldClose();
+}
+
+char* Nrt_IWindowingDevice_GetWindowTitle(NrtIWindowingDeviceHandle device)
+{
+    auto title = reinterpret_cast<IWindowingDevice*>(device)->GetWindowTitle();
```

Comment:
If this doesn't throw in C++ land, can we not use `.c_str()` to handle the allocation for us and return a `const char*` instead?

Reply:
That won't work. `GetWindowTitle` returns a `std::string` by value which copies the string onto the stack (this is what happens in this line). So using `c_str()` here will return a pointer to a string that's going to deleted as soon as the function ends.

Reply:
Ah, now I remember.

I think this is going to require more thought, might be out of scope for this PR otherwise I think this change will get held up too long.